ci: support private artifacts in fxci integration tests

ahal opened the pull request.

ahal commented:
/taskcluster integration

ahal commented:
Sweet, looks like this is finally working!

Some things on the TODO list:
* tests
* using cached tasks for the `firefoxci-artifact` kind

But these are things that could be done in a follow-up, so I'll flag for review now and incoporate those later (either in this PR or another).

hneiva commented on taskcluster/scripts/fetch-firefoxci-artifacts.py:
```diff
@@ -0,0 +1,53 @@
+credentials = result["secret"]
+assert isinstance(credentials, dict)
+
+options = {"rootUrl": FIREFOXCI_ROOT_URL, "credentials": credentials}
```
is this intended behavior? `options` is already set up above with `rootUrl`, and used to fetch credentials, and this one for the queue. Is there a use case where they would be using different `rootUrl`?

bhearsum replied:
AIUI, it first talks to staging to fetch the secret that contains the credentials for the Firefox CI cluster.

This might be a bit less confusing if there were two distinct `options` and/or a comment about the above.

ahal replied:
I'll add a comment.

hneiva commented on taskcluster/scripts/fetch-firefoxci-artifacts.py:
```diff
@@ -0,0 +1,53 @@
+
+task_id = os.environ["FETCH_FIREFOXCI_TASK_ID"]
+artifacts = json.loads(os.environ["FETCH_FIREFOXCI_ARTIFACTS"])
+for artifact in artifacts:
```
could we async download the artifacts? (maybe an improvement for another PR)

ahal replied:
Good idea, though it's probably pretty rare that we'd be fetching more than one artifact from a task anyway. Though thinking about it, we'd need to depend on aiohttp just for this, so I think I'll stick with synchronous for now.

hneiva reviewed (approved):
couple of minor questions, but LGTM otherwise

bhearsum reviewed (changes requested):
I'm requesting changes for the hardcoded artifact name, which I assume can be adjusted. If I'm missing something there I'll happily switch this to Approved.

ahal commented:
I refactored the script to use a couple functions and added an assertion to ensure we're only running this script from the staging instance. Hope that makes it more clear!

bhearsum reviewed (approved):
That's a nice refactor of the script; much clearer now!
